# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderCompositorD3D11SWGL.h
# Commit: 9d637c05af7f
# Full Hash: 9d637c05af7f8c39af78b290335ca55994849ee8
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1673983
# File Overlap Count: 2
# Description:
#   Bug 1673983 - Add RenderTextureHostSWGL support to D3D11 RenderTextureHosts. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D95799
# ==============================================================================

diff -r ea88b243717a -r 9d637c05af7f gfx/webrender_bindings/RenderCompositorD3D11SWGL.h
--- a/gfx/webrender_bindings/RenderCompositorD3D11SWGL.h	Fri Nov 06 00:00:47 2020 +0000
+++ b/gfx/webrender_bindings/RenderCompositorD3D11SWGL.h	Fri Nov 06 00:01:36 2020 +0000
@@ -49,6 +49,7 @@
   layers::WebRenderCompositor CompositorType() const override {
     return layers::WebRenderCompositor::D3D11;
   }
+  RenderCompositorD3D11SWGL* AsRenderCompositorD3D11SWGL() { return this; }
 
   // Interface for wr::Compositor
   CompositorCapabilities GetCompositorCapabilities() override;
@@ -95,6 +96,8 @@
     int32_t mY;
   };
 
+  ID3D11Device* GetDevice() { return mCompositor->GetDevice(); }
+
  private:
   RefPtr<layers::CompositorD3D11> mCompositor;
   void* mContext = nullptr;